lib/spear/connection/request.ex

```diff
@@ -108,10 +121,17 @@ defmodule Spear.Connection.Request do
        when finished in [:done, :halted] do
     request = put_in(request.status, :done)
 
+    messages =
+      if request.rpc.request_stream? and request.rpc.response_stream? do
+        message_buffer
+      else
+        [:eof | message_buffer]
+      end
```

Comment:
this is the important part of the change to the connection: making sure to never send `:eof` (which ends the request stream) for a bi-directional stream rpc like persistent.Read

```protobuf
rpc Read (stream ReadReq) returns (stream ReadResp);
```